### PR TITLE
Defer obtaining URL for Customizer preview until AMP Customizer is accessed

### PR DESCRIPTION
**Pull request by westonruter**
## Summary

This addresses a [support forum topic](https://wordpress.org/support/topic/wp_query-get_posts-slow-query-when-add-edit-post/):

> When working in the backend using the Classic Editor, we have noticed significant slow queries from the following call related to the AMP plugin.
>
> We have 20k posts and 500k comments.
>
> image
>
> After reviewing previous threads I can confirm AMP developer tools is deactivated. Any other thoughts on how to reduce or eliminate this call on the admin backend? We don’t need to use any of the settings or meta boxes for AMP on a per-post basis, is there an option to not load this call at all?

The issue here is the AMP Customizer link which is added under the Appearance menu when Reader mode is selected:

> image

In order to have the Customizer open with an AMP page in the preview, we do a query to find an AMP page and then supply it in the `url` parameter to `customize.php`. However, we don't have to supply this URL at this point. We can defer lookup up the URL until the Customizer is actually accessed. We can omit the `url` parameter and instead supply a custom one like `amp_preview`. When that is supplied when accessing `customize.php` and Reader mode is active, in the `customize_controls_init` action we can at that point do the lookup for an AMP page and pass it to `WP_Customize_Manager::set_preview_url()`. In this way the preview URL will be determined just in time. 

This only applies in Reader mode to the aforementioned AMP admin menu item under Appearance and also to the “Customize Reader Theme” link on the AMP Settings screen:

> image

----

This PR also fixes a deprecation notice which was something I missed in #5793.

## Checklist

- [ ] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [ ] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

File: includes/admin/class-amp-template-customizer.php [hunk 1]
```diff
@@ -123,6 +126,18 @@ public static function init( WP_Customize_Manager $wp_customize, ReaderThemeLoad
 		return $self;
 	}
 
+	/**
+	 * Set the preview URL when using a Reader theme if the AMP preview permalink was requested and no URL was provided.
+	 */
+	public function set_reader_preview_url() {
```
Comment (westonruter) [inline includes/admin/class-amp-template-customizer.php:132]: I should add a test for this as well.

File: includes/amp-post-template-functions.php [hunk 2]
```diff
@@ -125,6 +125,37 @@ function amp_post_template_add_styles( $amp_template ) {
...
+	 * @since 0.4
+	 *
+	 * @param array   $analytics An associative array of the analytics entries we want to output. Each array entry must have a unique key, and the value should be an array with the following keys: `type`, `attributes`, `script_data`. See readme for more details.
+	 * @param WP_Post $post      The current post.
+	 */
+	$analytics = apply_filters( 'amp_post_template_analytics', $analytics, get_queried_object() );
+
+	return $analytics;
+}
+
```
Comment (westonruter) [inline includes/amp-post-template-functions.php:158]: This was simply moved from `includes/admin/functions.php`

**Review by pierlon — APPROVED** [review]
LGTM :shipit:.

**Comment by pierlon** [comment]
**QA Passed**

With Reader mode enabled, I verified with Query Monitor that the query to find the latest AMP page is no longer being run:

image

There is no `url` query parameter attached to the AMP Customizer link:

image

Clicking on the customizer link does indeed display latest AMP page in the preview:

image

**Pull request merged**